### ISSUE TITLE
mssql: regression 0.2.21: databaseName is undefined with database url

### ISSUE THREAD

**Issue by kraiz** [typeorm/typeorm#5341]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[x] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `@next (0.3.0-rc2)` error
[x] `latest (0.2.22)` error
[x] `0.2.21` error
[ ] `0.2.20` works fine

**Steps to reproduce or a small repository showing the problem:**

I like to configure the database by `url`. This worked quite well until 0.2.21. Starting with this version I need to set the `database` attribute as well. Otherwise it fails somewhere inside the migration detection where it database name is `undefined`. Details below:

* Works fine with version `0.2.20`:
```
// ormconfig.json
{
  type: 'mssql',
  url: 'mssql://user:password@database:1433/dbname',
  ...
}
```

* Gives the following error with version `0.2.21`:
```
// ormconfig.json
{
  type: 'mssql',
  url: 'mssql://user:password@database:1433/dbname',
  ...
}
```

```
query: SELECT * FROM "undefined"."INFORMATION_SCHEMA"."TABLES" WHERE "TABLE_NAME" = 'migrations' AND "TABLE_SCHEMA" = SCHEMA_NAME()
query failed: SELECT * FROM "undefined"."INFORMATION_SCHEMA"."TABLES" WHERE "TABLE_NAME" = 'migrations' AND "TABLE_SCHEMA" = SCHEMA_NAME()
error: RequestError: Invalid object name 'undefined.INFORMATION_SCHEMA.TABLES'.
    at handleError (/plugins/node_modules/mssql/lib/tedious/request.js:366:15)
    at Connection.emit (events.js:210:5)
    at Parser.<anonymous> (/plugins/node_modules/tedious/lib/connection.js:800:12)
    at Parser.emit (events.js:210:5)
    at Parser.<anonymous> (/plugins/node_modules/tedious/lib/token/token-stream-parser.js:33:14)
    at Parser.emit (events.js:210:5)
    at addChunk (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_readable.js:297:12)
    at readableAddChunk (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_readable.js:279:11)
    at Parser.Readable.push (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_readable.js:240:10)
    at Parser.Transform.push (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_transform.js:139:32) {
  code: 'EREQUEST',
  originalError: Error: Invalid object name 'undefined.INFORMATION_SCHEMA.TABLES'.
      at handleError (/plugins/node_modules/mssql/lib/tedious/request.js:364:19)
      at Connection.emit (events.js:210:5)
      at Parser.<anonymous> (/plugins/node_modules/tedious/lib/connection.js:800:12)
      at Parser.emit (events.js:210:5)
      at Parser.<anonymous> (/plugins/node_modules/tedious/lib/token/token-stream-parser.js:33:14)
      at Parser.emit (events.js:210:5)
      at addChunk (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_readable.js:297:12)
      at readableAddChunk (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_readable.js:279:11)
      at Parser.Readable.push (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_readable.js:240:10)
      at Parser.Transform.push (/plugins/node_modules/tedious/node_modules/readable-stream/lib/_stream_transform.js:139:32) {
    info: ErrorMessageToken {
      name: 'ERROR',
      event: 'errorMessage',
      number: 208,
      state: 1,
      class: 16,
      message: "Invalid object name 'undefined.INFORMATION_SCHEMA.TABLES'.",
      serverName: '937b2ad281ae',
      procName: '',
      lineNumber: 1
    }
  },
  name: 'RequestError',
  number: 208,
  lineNumber: 1,
  state: 1,
  class: 16,
  serverName: '937b2ad281ae',
  procName: '',
  precedingErrors: []
}
```

* Works fine in version `0.2.21`:
```
// ormconfig.json
{
  type: 'mssql',
  url: 'mssql://user:password@database:1433/dbname',
  database: 'dbname'   // <-- this fixes the problem but is not DRY
  ...
}
```

**Comment by imnotjames** [comment]
It definitely looks like the code that translates from a URL to the driver options has been commented out: https://github.com/typeorm/typeorm/blob/master/src/driver/sqlserver/SqlServerDriver.ts#L224

**Comment by imnotjames** [comment]
Ah, this should be fixed as of 0.2.26 !
